pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/array/ArrayAggDistinctDoubleFunction.java

```diff
@@ -39,14 +39,15 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
       Map<ExpressionContext, BlockValSet> blockValSetMap) {
     BlockValSet blockValSet = blockValSetMap.get(_expression);
     double[] value = blockValSet.getDoubleValuesSV();
-    DoubleOpenHashSet valueArray = new DoubleOpenHashSet(length);
+    DoubleOpenHashSet valueSet = aggregationResultHolder.getResult() != null ? aggregationResultHolder.getResult()
+        : new DoubleOpenHashSet(length);
```

Comment:
(minor) Suggest following the same convention as other aggregates, same for other places

```suggestion
    DoubleOpenHashSet valueSet = aggregationResultHolder.getResult();
    if (valueSet == null) {
      valueSet = new DoubleOpenHashSet(length);
      aggregationResultHolder.setValue(valueSet);
    }
```

Reply:
We're updating `valueSet` inside the lambda function for `forEachNotNull` so it needs to be effectively final. In your suggested pattern, we'd need to unnecessarily copy it to an effectively final variable (no impact in terms of perf, but looks ugly). IMO, the current way looks cleaner overall so I'm gonna merge this. If you disagree, I'd be happy to open a follow-up PR.